jamazon/main.js

```diff
@@ -175,17 +211,66 @@ function showHidden(view) {
   })
 }
 
+const $appCatalog = document.querySelector('[data-view="catalog"]')
+const $appDetails = document.querySelector('[data-view="details"]')
+const $cart = document.querySelector('[data-view="cart"]')
+const $appCart = document.querySelector('#cart')
+
+$appCatalog.addEventListener('click', function (element) {
+  const item = element.target.parentNode.closest('.card')
+  if (item) {
+    const itemId = item.getAttribute('data-item-id')
+    app.view = 'details'
+    app.details.item = app.catalog.items[itemId - 1]
```

Comment:
It's not good to rely on the item's position to find it. Otherwise, there's no reason for it to have an id. You should have a function that takes all the items and an item id and returns you the matching item.